cmd/cli/certificate_rotate.go

```diff
@@ -0,0 +1,352 @@
+		if err != nil {
+			return nil, fmt.Errorf("unable to verify that MeshRootCertificate [%s]. Error: %v", existing.Name, err)
+		}
+
+		if existing.Spec.Intent != v1alpha2.InactiveIntent {
+			return nil, fmt.Errorf("MRC provided must be in Inactive role")
+		}
+
+		return existing, nil
+	}
+
+	if r.mrcFilePath != "" {
+		fmt.Fprintf(r.out, "using file %s\n", r.mrcFilePath)
+		decode := scheme.Codecs.UniversalDeserializer().Decode
+		file, err := os.ReadFile(r.mrcFilePath)
+		if err != nil {
+			return nil, err
+		}
+		obj, _, err := decode([]byte(file), nil, nil)
+		if err != nil {
+			return nil, err
+		}
+
+		newMRC, ok := obj.(*v1alpha2.MeshRootCertificate)
+		if !ok {
+			return nil, fmt.Errorf("file provided is not recognized as MeshRootCertificate")
+		}
+		if newMRC.Spec.Intent != v1alpha2.InactiveIntent {
+			return nil, fmt.Errorf("file provided must be a MeshRootCertificate in the Inactive role")
+		}
+
+		existing, err := r.configClient.ConfigV1alpha2().MeshRootCertificates(settings.Namespace()).Get(context.Background(), newMRC.Name, metav1.GetOptions{})
+		if err != nil {
+			return nil, fmt.Errorf("unable to verify that MeshRootCertificate [%s] doesn't already exist. Error: %v", newMRC.Name, err)
+		}
+		if existing != nil {
+			return nil, fmt.Errorf("cannot use MeshRootCertificate [%s] as it already exists in cluster", newMRC.Name)
+		}
+
+		return r.configClient.ConfigV1alpha2().MeshRootCertificates(settings.Namespace()).Create(context.Background(), newMRC, metav1.CreateOptions{})
+	}
+
+	return r.createFromExisting(mrc)
+}
+
+func (r *rotateCmd) createFromExisting(mrc *v1alpha2.MeshRootCertificate) (*v1alpha2.MeshRootCertificate, error) {
+	switch {
+	case mrc.Spec.Provider.Tresor != nil:
+		return r.createTresorFromExisting(mrc)
+	case mrc.Spec.Provider.Vault != nil:
+		// since we can't create the secrets in vault we require them to create a file
+		return nil, fmt.Errorf("please use -file or -meshrootcertificate option to provide a MeshRootCertificate Vault provider")
+	case mrc.Spec.Provider.CertManager != nil:
+		// since we require them to set up cert-manager.io ahead of time we can create it for them
+		return nil, fmt.Errorf("please use -file or -meshrootcertificate option to provide a MeshRootCertificate for CertManager provider")
+	default:
+		return nil, fmt.Errorf("unknown certificate provider: %+v", mrc.Spec.Provider)
+	}
+}
+
+func generateName() (string, string) {
+	var chars = []rune("abcdefghijklmnopqrstuvwxyz1234567890")
+
+	s := make([]rune, 8)
+	for i := range s {
+		num, _ := rand.Int(rand.Reader, big.NewInt(int64(len(chars))))
+		s[i] = chars[num.Int64()]
+	}
+	return fmt.Sprintf("%s-%x", constants.DefaultMeshRootCertificateName, string(s)), string(s)
+}
+
+func (r *rotateCmd) createTresorFromExisting(mrc *v1alpha2.MeshRootCertificate) (*v1alpha2.MeshRootCertificate, error) {
```

Comment:
this method doesn't necessarily need to know that the mrc passed in is pre-existing, so I would consider changing the name to just `createTresorMRC`